modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/DateTimeInput.es.js

```diff
@@ -43,65 +47,82 @@ class DateTimeInput extends React.Component {
 		return returnVal;
 	}
 
-	_handleDateChange = (event) => {
-		const value = event.target.value;
-
+	_handleDateChange = (value) => {
 		this.setState({value});
 	};
 
-	_handleDateBlur = (event) => {
-		const date = dateFns.format(event.target.value, INPUT_DATE_FORMAT);
+	_handleExpandedChange = (expandedState) => {
+		this.setState({expanded: expandedState});
 
-		if (date !== 'Invalid Date') {
-			this.setState(
-				{
-					value: date,
-				},
-				() => {
-					this.props.onChange({
-						type: PROPERTY_TYPES.DATE_TIME,
-						value: dateFns
-							.parse(date, INPUT_DATE_FORMAT)
-							.toISOString(),
-					});
-				}
-			);
+		if (expandedState === false) {
+			this._saveDateTimeValue();
 		}
-		else {
-			const resetDate = dateFns.format(new Date(), INPUT_DATE_FORMAT);
+	};
+
+	_saveDateTimeValue = () => {
+		const dateInput = dateFns.format(this.state.value, INPUT_DATE_FORMAT);
 
+		if (this.previousValue !== dateInput) {
 			this.setState(
 				{
-					value: resetDate,
+					value:
+						dateInput !== 'Invalid Date'
+							? dateInput
+							: dateFns.format(new Date(), INPUT_DATE_FORMAT),
 				},
 				() => {
 					this.props.onChange({
-						type: PROPERTY_TYPES.DATE_TIME,
-						value: dateFns
-							.parse(resetDate, INPUT_DATE_FORMAT)
-							.toISOString(),
+						type: this.props.propertyType,
+						value:
+							this.props.propertyType === 'date-time'
+								? dateFns
+										.parse(
+											this.state.value,
+											OUTPUT_DATE_FORMAT
+										)
+										.toISOString()
+								: dateFns.format(
+										this.state.value,
+										OUTPUT_DATE_FORMAT
+								  ),
 					});
 				}
 			);
 		}
+		this.previousValue = dateInput;
```

Comment:
Maybe you can store this previous value in the state and update it in the same setState call. This way the code will look cleaner.

Reply:
Understood, thank you